src/admin/resolvers/mutations.ts

```diff
@@ -439,6 +439,7 @@ export async function collectionImageUpload(
   { s3, db }: { s3: S3; db: PrismaClient }
 ) {
   const { image, ...imageData } = data;
+  await data.image.promise;
```

Comment:
but what...is this doing? could this have been `await image.promise;`, or is there something special about targeting it through the `data` object?

would love a comment on this line at some point. 😀

Reply:
@jpetto it probably could have been `await image.promise` this was the weirdest thing ever. I tried `await image` which did not work, but when i connected to the debugger I noticed that there was a promise object that was pending on the image object. And once i awaited it, the image populated with the data we were expecting. 

Why is this different thru federation? Why does curated-corpus do it differently?? Mysteries of the universe!

Reply:
![comfy-nervous](https://user-images.githubusercontent.com/317532/151588815-d314d0fa-2bbe-4b77-bee5-cc6d9cd21438.gif)